### PR TITLE
Refactor filestormanager tests

### PR DESCRIPTION
**Pull request by geirst**
@vekterli and / or @baldersheim please review

### CODE REVIEW

File: storage/src/tests/persistence/filestorage/filestormanagertest.cpp [hunk 1]
```diff
@@ -218,6 +219,57 @@ struct TestFileStorComponents {
...
+    {
+        top.push_back(std::unique_ptr<StorageLink>(dummyManager));
+        top.open();
+
+        metrics.initDiskMetrics(loadTypes.getMetricLoadTypes(), 1, threadsPerDisk);
+
+        filestorHandler = std::make_unique<FileStorHandlerImpl>(messageSender, metrics, test._node->getComponentRegister());
+        filestorHandler->setGetNextMessageTimeout(50ms);
+    }
+    ~FileStorHandlerComponents() {}
```
Comment (vekterli) [inline storage/src/tests/persistence/filestorage/filestormanagertest.cpp:246]: Consider `= default`

File: storage/src/tests/persistence/filestorage/filestormanagertest.cpp [hunk 2]
```diff
@@ -218,6 +219,57 @@ struct TestFileStorComponents {
...
+        persistenceHandler =
+                std::make_unique<PersistenceHandler>(test._node->executor(), component, cfg,
+                                                     test._node->getPersistenceProvider(),
+                                                     *filestorHandler, bucketOwnershipNotifier,
+                                                     *metrics.disk->threads[0]);
+    }
+    std::unique_ptr<DiskThread> make_disk_thread() {
+        return createThread(*persistenceHandler, *filestorHandler, component);
+    }
+    ~PersistenceHandlerComponents() {}
```
Comment (vekterli) [inline storage/src/tests/persistence/filestorage/filestormanagertest.cpp:270]: Consider `= default`

**Review by vekterli — APPROVED** [review]
:+1:

File: storage/src/tests/persistence/filestorage/filestormanagertest.cpp (hunk 1 above)
Comment (geirst) [reply storage/src/tests/persistence/filestorage/filestormanagertest.cpp:246]: That was not enough in this case.

File: storage/src/tests/persistence/filestorage/filestormanagertest.cpp (hunk 2 above)
Comment (geirst) [reply storage/src/tests/persistence/filestorage/filestormanagertest.cpp:270]: That was not enough in this case.

Comment (vekterli) [reply storage/src/tests/persistence/filestorage/filestormanagertest.cpp:270]: Move `PersistenceHandlerComponents::~PersistenceHandlerComponents() = default;` outside the class itself and the inliner becomes happy again

Comment (baldersheim) [reply storage/src/tests/persistence/filestorage/filestormanagertest.cpp:270]: Yes, that is the correct approach. = default in class definition applies the same inliner heuristics as not defining the destructor. Defining and empty destructor with {} changes the heuristics, while doing outside the class definitions avoids the problem.

Comment (geirst) [reply storage/src/tests/persistence/filestorage/filestormanagertest.cpp]: Fixed.

**Pull request merged**